packages/integrations/node/test/api-route.test.js

```diff
@@ -89,4 +89,23 @@ describe('API routes', () => {
 		let [out] = await done;
 		expect(new Uint8Array(out.buffer)).to.deep.equal(expectedDigest);
 	});
+
+	it('Can bail on streaming', async () => {
+		const { handler } = await import('./fixtures/api-route/dist/server/entry.mjs');
+		let { req, res, done } = createRequestAndResponse({
+			url: '/streaming',
+		});
+
+		let locals = { cancelledByTheServer: false };
+
+		handler(req, res, () => {}, locals);
+		req.send();
+
+		await new Promise((resolve) => setTimeout(resolve, 500));
+		res.emit("close");
```

Comment:
This line might need another look. I wasn't able to mock actual disconnection which would have emitted the event realistically. I don't know enough about node to take it further.

Reply:
I think this fine 

>Indicates that the response is completed, or its underlying connection was terminated prematurely (before the response completion).

https://nodejs.org/api/http.html#event-close_2